scripts/templates/ldrddi.cpp.mako

```diff
@@ -135,11 +135,21 @@ namespace loader
         %else:
         %if re.match(r"\w+ImageDestroy$", th.make_func_name(n, tags, obj)):
         // remove the handle from the kernel arugment map
-        context->image_handle_map.erase(reinterpret_cast<ze_image_object_t*>(hImage));
+        {
+            std::lock_guard<std::mutex> lock(context->image_handle_map_lock);
+            if( context->image_handle_map.find(reinterpret_cast<ze_image_object_t*>(hImage)) != context->image_handle_map.end() ) {
```

Comment:
Why do you avoid erasing it if it's the only element remaining?

Reply:
you are right, I think I was being paranoid, I can remove the find loop.

Reply:
fixed in latest.